sympy/matrices/matrices.py

```diff
@@ -1594,6 +1596,12 @@ def __pow__(self, num):
                     return self._matrix_pow_by_recursion(num)
             return self._matrix_pow_by_recursion(num)
         elif isinstance(num, (Expr, float)):
+            num = sympify(num)
+            if (num >= 0) == True and not num % 1 == 0:
+                z = self.jordan_form()[1]
```

Comment:
I think that should be done in `_matrix_pow_by_jordan_blocks` after the Jordan cells have been found. That way there would be no duplicate computation.